Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -86,10 +86,14 @@ private class ParameterSetNames
 
         /// <summary>
         /// <para type="description">
-        /// The ID of the dataset that you want to get a descriptor object for.
+        /// The ID of the dataset that you want to get a descriptor object for. This field also accepts 
+        /// DatasetData objects so they can be mapped to full Dataset objects.
         /// </para>
         /// </summary>
-        [Parameter(Mandatory = true, Position = 0, ParameterSetName = ParameterSetNames.Get)]
+        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.Get)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(DatasetList.DatasetsData),
+            Property = nameof(DatasetList.DatasetsData.DatasetReference))]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(DatasetReference), Property = nameof(DatasetReference.DatasetId))]
         public string Dataset { get; set; }
```

Comment:
Alright, so that's looking like it's going to take a lot more thought and code fixing, because the parameter sets are going to get kinda messy (with all of the data transforms and stuff).  I'm going to just commit this and then open up a new PR for changes to address this issue once I'm done with some of the design doc changes and demo prep that I want to get done for monday.  plus this PR is huge, and it'll help to be able to focus in on these few dataset issues in their own PR instead of in a Table PR.